recipes/create-dmg/all/test_package/conanfile.py

```diff
@@ -0,0 +1,9 @@
+from conans import ConanFile, tools
+
+
+class TestPackageConan(ConanFile):
+    settings = "os"
```

Comment:
```suggestion
    settings = "os", "arch", "compiler", "build_type"
```
Use full settings always